R/ResultsDataModelQueries.R

```diff
@@ -0,0 +1,977 @@
+#' Returns a data frame (tibble) with results that conform to incidence_rate
+#' table in Cohort Diagnostics results data model.
+#'
+#' @export
+getResultsFromIncidenceRate <- function(dataSource,
+                                        cohortIds,
+                                        databaseIds) {
+  data <- getDataFromResultsDatabaseSchema(
+    dataSource,
+    cohortIds = cohortIds,
+    databaseIds = databaseIds,
+    dataTableName = 'incidenceRate'
+  )
+  return(data)
+}
+
+#' Returns data from inclusion_rule_stats table of Cohort Diagnostics results data model
+#'
+#' @description
+#' Returns data from inclusion_rule_stats table of Cohort Diagnostics results data model
+#'
+#' @template DataSource
+#'
+#' @template CohortIds
+#'
+#' @template DatabaseIds
+#'
+#' @return
+#' Returns a data frame (tibble) with results that conform to inclusion_rule_stats
+#' table in Cohort Diagnostics results data model.
+#'
+#' @export
+getResultsFromInclusionRuleStatistics <- function(dataSource,
+                                                  cohortIds,
+                                                  databaseIds) {
+  data <- getDataFromResultsDatabaseSchema(
+    dataSource,
+    cohortIds = cohortIds,
+    databaseIds = databaseIds,
+    dataTableName = 'inclusionRuleStats'
+  )
+  return(data)
+}
+
+
+#' Returns data from index_event_breakdown table of Cohort Diagnostics results data model
+#'
+#' @description
+#' Returns data from index_event_breakdown table of Cohort Diagnostics results data model
+#'
+#' @template DataSource
+#'
+#' @template CohortIds
+#'
+#' @template DatabaseIds
+#'
+#' @return
+#' Returns a data frame (tibble) with results that conform to index_event_breakdown
+#' table in Cohort Diagnostics results data model.
+#'
+#' @export
+getResultsFromIndexEventBreakdown <- function(dataSource,
+                                              cohortIds,
+                                              databaseIds) {
+  data <- getDataFromResultsDatabaseSchema(
+    dataSource,
+    cohortIds = cohortIds,
+    databaseIds = databaseIds,
+    dataTableName = 'indexEventBreakdown'
+  )
+  return(data)
+}
+
+
+#' Returns data from concept table of Cohort Diagnostics results data model
+#'
+#' @description
+#' Returns data from concept table of Cohort Diagnostics results data model
+#'
+#' @template DataSource
+#'
+#' @template ConceptIds
+#'
+#' @template VocabularyDatabaseSchema
+#'
+#' @return
+#' Returns a data frame (tibble) with results that conform to concept
+#' table in Cohort Diagnostics results data model.
+#'
+#' @export
+getConceptDetails <- function(dataSource = .GlobalEnv,
+                              vocabularyDatabaseSchema = NULL,
+                              conceptIds) {
+  table <- 'concept'
+  if (!is.null(vocabularyDatabaseSchema) &&
+      is(dataSource, "environment")) {
+    warning(
+      'vocabularyDatabaseSchema provided for function getConceptDetails in non database mode. This will be ignored.'
```

Comment:
Would you mind consistently using double quotes in your R code?

Reply:
absolutely